### PR TITLE
cleanup(userspace/libsinsp): drop user and group infos embedded in threadinfo

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area libsinsp

**Does this PR require a change in the driver versions?**

**What this PR does / why we need it**:

Only store `uid`, `gid` and `loginuid` info.
Storing full info was done to workaround scenarios like:
* thread X starts with user 1000 "foo"
* user 1000 gets deleted while thread X runs
* user 1000 gets recreated with name "bar"
* extracting user from thread now will give "bar"; with previous impl it would have given "foo".

Basically, we are now respecting what unix expects us to do, since user 1000 will now be mapped to "bar".

Moreover, since threadinfo is used many times, not embedding those bytes saves us lots of memory.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: userspace/libsinsp/parsers.cpp [hunk 1]
```diff
@@ -1349,13 +1349,6 @@ void sinsp_parser::parse_clone_exit_caller(sinsp_evt *evt, int64_t child_tid) {
 		return;
 	}
 
-	/* Refresh user / loginuser / group */
-	if(new_child->m_container_id.empty() == false) {
```
Comment (FedeDP) [inline userspace/libsinsp/parsers.cpp:1353]: No need to refresh user/group anymore when container id changes.
User group manager does now attach the `subscribe_on_new_container` listener on the container manager to cache all new containers users.

File: userspace/libsinsp/threadinfo.cpp [hunk 2]
```diff
@@ -511,88 +514,82 @@ void sinsp_threadinfo::init(scap_threadinfo* pi) {
...
+}
+
+std::string sinsp_threadinfo::get_exe() const {
+	return m_exe;
+}
+
+std::string sinsp_threadinfo::get_exepath() const {
+	return m_exepath;
+}
+
```
Comment (FedeDP) [inline userspace/libsinsp/threadinfo.cpp:532]: No changes here.

File: userspace/libsinsp/threadinfo.cpp [hunk 3]
```diff
@@ -511,88 +514,82 @@ void sinsp_threadinfo::init(scap_threadinfo* pi) {
...
+	static scap_groupinfo grp = {};
+	grp.gid = m_gid;
+	strlcpy(grp.name, m_gid == 0 ? "root" : "<NA>", sizeof(grp.name));
+	return &grp;
 }
 
-std::string sinsp_threadinfo::get_exepath() const {
-	return m_exepath;
+scap_userinfo* sinsp_threadinfo::get_loginuser() const {
+	auto user = m_inspector->m_usergroup_manager.get_user(m_container_id, m_loginuid);
```
Comment (Andreagit97) [inline userspace/libsinsp/threadinfo.cpp:582]: this is the same code we use in `set_user` apart from a different id, can we create an helper method

File: userspace/libsinsp/threadinfo.cpp [hunk 4]
```diff
@@ -511,88 +514,82 @@ void sinsp_threadinfo::init(scap_threadinfo* pi) {
...
-		}
-	} else {
-		// No need to set name/homedir/shell, the default values from
-		// sinsp_userinfo are going to be used.
-		m_group.set_gid(gid);
-	}
-	m_user.set_gid(m_group.gid());
 }
 
 void sinsp_threadinfo::set_loginuser(uint32_t loginuid) {
```
Comment (Andreagit97) [inline userspace/libsinsp/threadinfo.cpp:552]: I would rename this into `set_loginuid`  since we are just doing it

File: userspace/libsinsp/parsers.cpp [hunk 5]
```diff
@@ -5075,9 +5075,9 @@ void sinsp_parser::parse_chroot_exit(sinsp_evt *evt) {
 		// if we happen to change container id
 		//
 		if(container_id != evt->get_tinfo()->m_container_id) {
-			evt->get_tinfo()->set_user(evt->get_tinfo()->m_user.uid());
-			evt->get_tinfo()->set_loginuser(evt->get_tinfo()->m_loginuser.uid());
-			evt->get_tinfo()->set_group(evt->get_tinfo()->m_group.gid());
+			evt->get_tinfo()->set_group(evt->get_tinfo()->m_gid);
+			evt->get_tinfo()->set_user(evt->get_tinfo()->m_uid);
+			evt->get_tinfo()->set_loginuser(evt->get_tinfo()->m_loginuid);
```
Comment (Andreagit97) [inline userspace/libsinsp/parsers.cpp:5080]: This line is doing `evt->get_tinfo()->m_loginuid = evt->get_tinfo()->m_loginuid;` or am i wrong?

**Review by Andreagit97 — COMMENTED** [review]
Thank you for the cleanup!

File: userspace/libsinsp/threadinfo.cpp (hunk 4 above)
Comment (Andreagit97) [reply userspace/libsinsp/threadinfo.cpp:552]: See the comment below and why it could be misleading

**Comment by FedeDP** [comment]
Thanks @Andreagit97 ! Pushed the changes addressing your review comments!

**Review by Andreagit97 — DISMISSED** [review]
/approve

**Comment by Andreagit97** [comment]
ops you missed a replace of `set_loginuid`

**Comment by FedeDP** [comment]
yup just saw that ahah

**Comment by FedeDP** [comment]
Fixed! Thanks!

**Review by jasondellaluce — APPROVED** [review]
